Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Writers/CloudFormationJsonWriter.cs

```diff
@@ -83,6 +83,7 @@ private void ProcessLambdaFunction(ILambdaFunctionSerializable lambdaFunction)
                 _jsonWriter.RemoveToken($"{propertiesPath}.Role");
             }
             
+            _jsonWriter.RemoveToken($"{lambdaFunctionPath}.Properties.Events");
```

Comment:
Looks like the logic is every time we do generation delete the events section and then recreate it. That will mess up users  that are customizing the event object in the template. For example we aren't modeling in code all of the authorization fields that can be done for the HTTP event. So they would need to update the field.

We need a better way of knowing when the attribute has been removed then we delete the event. Possibly putting more state in the metadata section of the AWS::Serverless::Function that identifies what fields were last synced.

Reply:
Agree with Norm, this too risky to delete the existing events and add every single time there is change. Only the relevant JSON blob must be update (not create delete).

Reply:
Added logic to surgically remove redundant attributes.